src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -284,29 +363,72 @@ public void InstallWorkloadPack(PackInfo packInfo, SdkFeatureBand sdkFeatureBand
         {
             try
             {
+                ReportPendingReboot();
+
                 // Determine the MSI payload package ID based on the host architecture, pack ID and pack version.
                 string msiPackageId = GetMsiPackageId(packInfo);
 
                 // Retrieve the payload from the MSI package cache. 
-                MsiPayload msiPayload = GetCachedMsiPayload(msiPackageId, packInfo.Version);
-                VerifyPackage(msiPayload);
-                DetectState state = Detect(msiPayload.Manifest.ProductCode, out string _);
-                PlanAndExecute(msiPayload, packInfo, state, InstallAction.Install);
+                MsiPayload msi = GetCachedMsiPayload(msiPackageId, packInfo.Version, offlineCache);
+                VerifyPackage(msi);
+                DetectState state = DetectPackage(msi, out Version installedVersion);
+                InstallAction plannedAction = PlanPackage(msi, state, InstallAction.Install, installedVersion, out _);
+                ExecutePackage(msi, plannedAction);
 
                 // Update the reference count against the MSI.
-                UpdateDependent(InstallRequestType.AddDependent, msiPayload.Manifest.ProviderKeyName, _dependent);
+                UpdateDependent(InstallRequestType.AddDependent, msi.Manifest.ProviderKeyName, _dependent);
             }
             catch (Exception e)
             {
                 LogException(e);
```

Comment:
Does this log the exception to the output or just to the log file? We'll want to make sure this error is displayed to the user so they know what went wrong.

Reply:
It logs the exception only to the log, but then throws it so that on the command side it get's into a GracefulException. The install log usually just adds more detail that we can use for investigating while on the UI we only show the message